src/Core/Settings.h

```diff
@@ -374,6 +374,9 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(UInt64, memory_profiler_step, (4 * 1024 * 1024), "Whenever query memory usage becomes larger than every next step in number of bytes the memory profiler will collect the allocating stack trace. Zero means disabled memory profiler. Values lower than a few megabytes will slow down query processing.", 0) \
     M(Float, memory_profiler_sample_probability, 0., "Collect random allocations and deallocations and write them into system.trace_log with 'MemorySample' trace_type. The probability is for every alloc/free regardless to the size of the allocation. Note that sampling happens only when the amount of untracked memory exceeds 'max_untracked_memory'. You may want to set 'max_untracked_memory' to 0 for extra fine grained sampling.", 0) \
     \
+    M(UInt64, guaranteed_memory_usage, 1_GiB, "", 0) \
+    M(UInt64, max_guaranteed_memory_usage_for_user, 1_GiB, "", 0) \
```

Comment:
Not sure if 'for user' is good to have as a number of megabytes. It depends very much on number of users registered or running in the system.
My intuition tells that it will not work at all - memory usage per query will not have any effect and every 2+ query will be treated as unguaranteed because there is for user guarantee.